plugins/org.yakindu.sct.generator.c/src/org/yakindu/sct/generator/c/submodules/MethodGenerator.xtend

```diff
@@ -0,0 +1,71 @@
+package org.yakindu.sct.generator.c.submodules
+
+import com.google.inject.Inject
+import org.yakindu.base.types.annotations.VisibilityAnnotations
+import org.yakindu.sct.generator.c.FlowCode
+import org.yakindu.sct.generator.c.extensions.Naming
+import org.yakindu.sct.generator.core.types.ICodegenTypeSystemAccess
+import org.yakindu.sct.model.sexec.Method
+import org.yakindu.sct.model.sexec.extensions.SExecExtensions
+import org.yakindu.sct.model.sexec.naming.INamingService
+
+class MethodGenerator {
+
+	@Inject protected extension VisibilityAnnotations
+	@Inject protected extension INamingService
+	@Inject protected extension ICodegenTypeSystemAccess
+	@Inject protected extension Naming
+	@Inject protected extension SExecExtensions
+	
+	@Inject protected extension FlowCode
+
+
+	def declarations(Iterable<Method> steps) '''
+		«FOR s : steps»
+			«s.declaration»
+		«ENDFOR»
+	'''
+	
+	
+	def declaration(Method it) '''
+		«IF ! body.comment.isNullOrEmpty»/*! «body.comment» */«ENDIF»
+		«declarationModifier»«typeSpecifier.targetLanguageName» «prefix»«shortName»(«scHandleDecl»«FOR p : parameters BEFORE ', ' SEPARATOR ', '»«IF p.varArgs»...«ELSE»const «p.typeSpecifier.targetLanguageName» «p.name.asIdentifier»«ENDIF»«ENDFOR»);
+		
+	'''
+	
+	def declarationModifier(Method it) {
+		if (it.isPublic)
+			'''extern '''
+		else 
+			'''static '''
+	}
+	
+
+	def implementation(Iterable<Method> methods) '''
+	 	«FOR m : methods»
+	 		«m.implementation»
+	 		
+	 	«ENDFOR»
+	 '''
+	
+	def implementation(Method it) '''
```

Comment:
merge master. This has been changed. See https://github.com/Yakindu/statecharts/pull/3135/files

Reply:
Is not missing - comments are part of method definitions. They are defined by the model sequencer and generated by MethodGenerator.